presto-main/src/main/java/com/facebook/presto/sql/analyzer/StatementAnalyzer.java

```diff
@@ -1348,11 +1356,27 @@ else if (expression instanceof LongLiteral) {
                     }
 
                     // otherwise, just use the expression as is
+                    boolean orderByOnExpression = false;
                     if (orderByExpression == null) {
                         orderByExpression = expression;
+                        orderByOnExpression = true;
                     }
 
                     ExpressionAnalysis expressionAnalysis = analyzeExpression(orderByExpression, sourceScope);
+                    if (orderByOnExpression) {
+                        // Try to resolve within orderByScope. If it is possible, then we warn - standard behavior differs from legacy one.
+                        for (Expression columnExpression : expressionAnalysis.getColumnReferences()) {
+                            Optional<ResolvedField> resolvedField = orderByScope.tryResolveField(columnExpression);
```

Comment:
@findepi Can you elaborate with example what could possibly go wrong? @sopel39 I analyze only orderByExpressions (loop over them), what is wrong here?